### ISSUE TITLE
Regridding artifacts in model-vs-model comparison

### ISSUE THREAD

**Issue by oksanaguba** [E3SM-Project/e3sm_diags#560]
The issue came up when i made model-vs-model climo for a run with ne30pg2 grid (new run, made with a mapping file below) against a run with ne30np4 grid (baseline, made with a different mapping file, also below). See blue contours 
https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/onguba/theta/theta-cpterms-jan27.144547.0002-0006/def/viewer/lat_lon/model_vs_model/t-850mb-global/ann.html

When i made climo against a ne30pg2 baseline run, the contours are gone:
https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/onguba/theta/theta-cpterms-jan27.144547-against-pg.0002-0006/def/viewer/lat_lon/model_vs_model/t-850mb-global/ann.html

Maps that i used for both grids are
map_ne30np4_to_fv128x256_aave.20160301.nc  
map_ne30pg2_to_cmip6_180x360_aave.20200201.nc

Since maps regular grids differ, may it be expected behavior?

**Comment by wlin7** [comment]
Hi @oksanaguba ,try to understand the issue.  Are the two baseline runs themselves actually different -- the middle panels of the two pages, one with ne30np4, the other with ne30pg2?

**Comment by oksanaguba** [comment]
yes, baselines differ, np4 and pg2, the run to investigate is pg2, thanks.

**Comment by wlin7** [comment]
Thanks. In that case, we would expect the results to be different; and the differences between the two difference plots (3rd panel in each) are due to differences in the baseline simulations. Therefore, the differences are a result of model configuration: the treatment of physics columns and the cumulative interaction between physics and dynamics over 6 years of simulations. Regridding (with e3sm_diag) itself is fine, with proper mapping files for the corresponding output data. But please further elaborate if I misunderstood the issue.

**Comment by oksanaguba** [comment]
Thanks, Wuyin, i agree it is not such a problem if one can use proper baselines. However, in this case it seems to me that blue contours appear close to places where one of files does not have data (which are marked in white). So, extrema are created for regions where one of files does not have data. If this is the case, this seems to be wrong behavior?  This example is also problematic because the destination grids do not match. 

In other words, to me it seems that blue contours are due to nonexistent data in the first two plots, not because there is difference between 2 simulations.

**Comment by wlin7** [comment]
Thanks for the explanation, Oksana. I went the wrong direction. I got your points now. Indeed the blue contour encircling the no-data areas are artifact. This could also happens when comparison with obs, when data to be compared are on different grids and/or have different fill values. The issue is internal to e3sm_diags on getting data on a common grid before computing differences.  

I am not sure though if it has been fixed in more recent versions. Pinging @chengzhuzhang and @forsyth2 .

**Comment by chengzhuzhang** [comment]
@wlin7 Thank you for chiming in. I really appreciated it. My immediate reaction on this problem is that the conservative regriding tool/algorithm from CDAT currently used in e3sm_diags have an issue handling masks. This behavior resemble that has been found here: https://github.com/E3SM-Project/e3sm_diags/issues/411. The team is working on a solution to replace the regriding tool. We will use this as a test case for the new algorithm.

**Comment by chengzhuzhang** [comment]
Updates: indeed replace with linear regriding method would remove the blue edge around masked data. Command line to replace:
```
e3sm_diags lat_lon --no_viewer --case_id 'model_vs_model' --sets 'lat_lon' --run_type 'model_vs_model' --variables 'T' --seasons 'ANN' --regions 'global' --regrid_tool 'esmf' --regrid_method 'linear' --plevs '850.0' --main_title 'T 850 mb ANN global' --backend 'mpl' --output_format 'png' --canvas_size_w '1212' --canvas_size_h '1628' --figsize '8.5' '11.0' --dpi '150' --arrows --contour_levels '240' '245' '250' '255' '260' '265' '270' '275' '280' '285' '290' '295' --test_title 'theta-cpterms-jan27.144547' --test_colormap 'cet_rainbow.rgb' --reference_colormap 'cet_rainbow.rgb' --diff_title 'Model - Observation' --diff_colormap 'diverging_bwr.rgb' --diff_levels '-5' '-4' '-3' '-2' '-1' '-0.5' '-0.25' '0.25' '0.5' '1' '2' '3' '4' '5' --multiprocessing --num_workers '64' --granulate 'variables' 'seasons' 'plevs' 'regions' --selectors 'sets' 'seasons' --reference_data_path '/lcrc/group/e3sm/ac.onguba/scratch/chrys/theta-clubb-fixer-default-climo/run/climoout.114060-0002-0006' --test_data_path '/home/ac.onguba/ess/theta-cpterms-jan27/run/climoout.144547-0002-0006' --results_dir '/lcrc/group/acme/public_html/diagnostic_output/ac.zhang40/test/e3sm_diags_issue540_linear
```
The only change is to replace --regrid_method 'linear' with --regrid_method 'conservative'
Will resolve this issue with regridder update.
